service/src/main/java/dk/dbc/opensearch/ServiceBean.java

```diff
@@ -0,0 +1,251 @@
+                cacheControl.setMaxAge(600);
+                return Response.ok(indexHtml.getBytes())
+                        .type(MediaType.TEXT_HTML_TYPE)
+                        .cacheControl(cacheControl)
+                        .build();
+            } else {
+                return badRequest("Support for uri parameters not implemented yet");
+            }
+        }
+    }
+
+    @GET
+    @Path("{path: .+}")
+    public Response docroot(@PathParam("path") String path, @Context HttpHeaders headers, @Context HttpServletRequest httpRequest) {
+        String peer = remoteIPAddress.ip(headers, httpRequest);
+        try (MDCLog mdc = MDCLog.mdc()
+                .withPeer(peer)) {
+            return ResourceStreamingOutput.docroot(path);
+        }
+    }
+
+    @POST
+    @Consumes(MediaType.APPLICATION_FORM_URLENCODED)
+    public Response post(@FormParam("req") String req, @Context HttpHeaders headers, @Context HttpServletRequest httpRequest) {
+        try (ByteArrayInputStream bis = new ByteArrayInputStream(req.getBytes(StandardCharsets.UTF_8))) {
+            try (BufferedInputStream is = new BufferedInputStream(bis, 1024)) {
+                is.mark(1024);
+                int c = is.read();
+                while (Character.isWhitespace(0)) {
+                    c = is.read();
+                }
+                if (c == -1)
+                    throw new IOException("Empty request");
+                is.reset();
+                if (c == '{')
+                    return processInputStream(headers, httpRequest, is, RequestParserJSON::new);
+                else
+                    return processInputStream(headers, httpRequest, is, RequestParser::new);
+            }
+        } catch (IOException ex) {
+            log.error("post (form): {}", ex.getMessage());
+            log.debug("post (form): ", ex);
+            return Response.serverError().build();
+        }
+    }
+
+    @POST
+    @Consumes({MediaType.APPLICATION_XML,
+               "application/soap+xml"})
+    @Produces({MediaType.APPLICATION_XML,
+               MediaType.APPLICATION_JSON})
+    public Response postXML(InputStream is, @Context HttpHeaders headers, @Context HttpServletRequest httpRequest) {
+        return processInputStream(headers, httpRequest, is, RequestParser::new);
+    }
+
+    @POST
+    @Consumes({MediaType.APPLICATION_JSON})
+    @Produces({MediaType.APPLICATION_XML,
+               MediaType.APPLICATION_JSON})
+    public Response postJSON(InputStream is, @Context HttpHeaders headers, @Context HttpServletRequest httpRequest) {
+        return processInputStream(headers, httpRequest, is, RequestParserJSON::new);
+    }
+
+    private Response processInputStream(HttpHeaders headers, HttpServletRequest httpRequest, InputStream is, RequestProvider requestProvider) {
+        String peer = remoteIPAddress.ip(headers, httpRequest);
+        try {
+            MDCLog mdc = mdc()
```

Comment:
```suggestion
            MDCLog mdc = MDCLog.mdc()
```
Så er det konsekvent med de andre steder `mdc()` bliver kaldt.